content/posts/2018-01-10-kotlin-null.md

```diff
@@ -0,0 +1,249 @@
+---
+title: "Kotlin: The Problem with `null`"
+date: 2018-01-10
+slug: kotlin-the-problem-with-null
+---
+
+I think almost everybody in the CS world knows about the [Null References: The Billion Dollar Mistake](https://www.infoq.com/presentations/Null-References-The-Billion-Dollar-Mistake-Tony-Hoare) talk.
+However, it does not really matter if a concept of `null` references a good idea or a really bad one.
+We live in the world where we just have to deal with `null` this way or another.
+Yep, just deal with it :sunglasses:. So let’s ask the “How” question instead of the “Why” one.
+
+# Enter Kotlin
+
+The drug of choice of mine at this point in time is Kotlin,
+but you cannot forget about Java when using Kotlin.
+Well, the JVM probably can be avoided by using Kotlin Native,
+but at least on Android you are stuck with the JVM as an intermediate bytecode provider.
+
+Kotlin was, and is a great language.
+Backed by a team at JetBrains behind the best IDE platform available,
+reasonable and pragmatic design decisions — what else to wish for?
+And, more importantly, it brought the concept of Null Safety
+to the production table for developers.
+
+All Java objects are essentially implicitly can be `null`,
+when Kotlin objects have to be declared nullable explicitly,
+otherwise the type system assumes them be to be non-nullable by default.
+
+Finally! We can use the JVM and forget about the `NullPointerException`!
+[Well, not exactly](https://en.wikipedia.org/wiki/Forgot_About_Dre).
+
+Let’s take a look at a very short Kotlin + RxJava sample.
+
+```kotlin
+Observable.just(null).subscribe()
+```
+
+This code will not be highlighted by the IDE, compiler will not show any warnings or errors and yet it will crash _in runtime_.
+Well, at least when using RxJava 2.1.8.
```

Comment:
I think you need to specify Kotlin compiler, IDE plugin and IDE versions as well

Reply:
Dude, where is my bug report, right?